mailbox/api/src/main/java/org/apache/james/mailbox/MessageManager.java

```diff
@@ -390,104 +392,180 @@ public Flags getFlags() {
             NO_COUNT
         }
 
-        /**
-         * Gets the UIDs of recent messages if requested or an empty
-         * {@link List} otherwise.
-         * 
-         * @return the uids flagged RECENT in this mailbox,
-         */
-        List<MessageUid> getRecent();
+        public static MailboxMetaData sensibleInformationFree(MailboxACL resolvedAcl, UidValidity uidValidity, boolean writeable, boolean modSeqPermanent) throws MailboxException {
+            ImmutableList<MessageUid> recents = ImmutableList.of();
+            MessageUid uidNext = MessageUid.MIN_VALUE;
+            ModSeq highestModSeq = ModSeq.first();
+            long messageCount = 0L;
+            long unseenCount = 0L;
+            MessageUid firstUnseen = null;
+            return new MailboxMetaData(
+                recents,
+                new Flags(),
+                uidValidity,
+                uidNext,
+                highestModSeq,
+                messageCount,
+                unseenCount,
+                firstUnseen,
+                writeable,
+                modSeqPermanent,
+                resolvedAcl);
+        }
+
+        private final long recentCount;
+        private final List<MessageUid> recent;
+        private final Flags permanentFlags;
+        private final UidValidity uidValidity;
+        private final MessageUid nextUid;
+        private final long messageCount;
+        private final long unseenCount;
+        private final MessageUid firstUnseen;
+        private final boolean writeable;
+        private final ModSeq highestModSeq;
+        private final boolean modSeqPermanent;
+        private final MailboxACL acl;
+
+        public MailboxMetaData(List<MessageUid> recent, Flags permanentFlags, UidValidity uidValidity, MessageUid uidNext, ModSeq highestModSeq, long messageCount, long unseenCount, MessageUid firstUnseen, boolean writeable, boolean modSeqPermanent, MailboxACL acl) {
+            this.recent = Optional.ofNullable(recent).orElseGet(ArrayList::new);
```

Comment:
why you want `arrayList` here?

Reply:
Because that is "just" code movement, bu I can make this immutable in another commit if you wish :-)

Reply:
ok